project_forecast_line/models/project_project.py

```diff
@@ -9,10 +9,24 @@ class ProjectProject(models.Model):
     def _update_forecast_lines_trigger_fields(self):
         return ["stage_id"]
 
+    def _get_written_computed_fields(self, values):
+        # get written and computed field names
+        task_computed = []
+        written_fields = list(values.keys())
+        tree = self.pool.field_triggers.get(self._fields[next(iter(written_fields))])
```

Comment:
Are you sure about this? It looks like you are only considering the first written field, and not any of the others.

Reply:
I see the problem, I'll correct.

Reply:
basically, I use implementation, but didn't pay attention to the details here:
https://github.com/odoo/odoo/blob/73de38e0e35f4ffb8069441cf11ebd48ff5aadb1/odoo/models.py#L5984

and maybe it would be better to use v16 variant
https://github.com/odoo/odoo/blob/99d7d0a7b875ffb1e5ce00cd7f3fc2c9e0359796/odoo/models.py#L5980